CMakeLists.txt

```diff
@@ -1244,17 +1244,7 @@ install(
 
 
 # Font files
-if(UNIX AND NOT APPLE)
-  # Ubuntu and OpenSans are already installed by package manager
-  install(
-    DIRECTORY
-      "${CMAKE_CURRENT_SOURCE_DIR}/res/fonts"
-    DESTINATION
-      "${MIXXX_INSTALL_DATADIR}"
-    PATTERN "OpenSans*" EXCLUDE
-    PATTERN "Ubuntu*" EXCLUDE
```

Comment:
I have original picked this solution, because it might be surprising that no font is installed on Linux only when a skin designer adds one for his skin. 
The issue will be very hard to discover, because it only happens when someone runs the installed Mixxx and not from the source tree.  This will unlikely happen during skin development. 

Is there an option to remove empty folders that we can use instead?

Reply:
I don't think so. The empty directory causes `namcap` warnings when building an Arch Linux package. Even if we add a new font, it should probably be installed via package manager anyway. Just in case, I'll ping @ronso0

Reply:
Yes, sir? tbh I don't understand the issue..
Are you questioning to deliver fonts with Mixxx?

Reply:
The issue is that it just works during development (only). How can we fail that early. Mmmm...

How about his (untested): 

file(GLOB MIXXX_FONTS "${CMAKE_CURRENT_SOURCE_DIR}/res/fonts/*")
list(FILTER MIXXX_FONTS INCLUDE REGEX <regex todo>)
if (MIXXX_FONTS)  
  install(
      FILES MIXXX_FONTS     
      DESTINATION "${MIXXX_INSTALL_DATADIR}")

Reply:
@ronso0 summarizing the discussion: We currently ship OpenSans and Ubuntu fonts in Windows and Apple. On Linux, the we don't ship them because they should be installed via package manager.

The old solution just created an empty fonts directory which causes warnings when packaging. The new solution just skips font install on Linux entirely.

@daschuer argues that if we add a new skin that uses another font and need to ship it on Linux because for some reason it's an obscure font that's not packaged then we might forget to re-enable font install for that font.

I think a) the chance that we add a new font that isn't installable via package manager is slim and b) if we do that we three now know what we'd add it to `CMakeLists.txt`.

Reply:
The DarkMetal skin has a the digital.otf font to install. If there is fonts directory it is obvious where to put the file. 

But yes, we can discuss the issue away. I am just afraid we will forget about it. 

The poor man solution would be to populate the fonts folder with a readme file ...

Reply:
Alright, good we talk it, I didn't install Mixxx via ppa for ages...
Neither OpenSans nor Ubuntu font are listed as dependency (mixxx 2.2.4 via Synaptics on Ubuntu 20.04).
How do users know "they should be installed via package manager"?

For LateNight 2.3 we _need_ Open Sans Semibold, besides for most labels it's also used in SVG icons.
just building 2.3 to check how it looks like with the sans serif replacement...meeeeh

Reply:
> How do users know "they should be installed via package manager"?

By listing them as a dependency. If they aren't, that's a packaging bug.

Reply:
> Neither OpenSans nor Ubuntu font are listed as dependency

They are installed by default, so no one will notice. However, I have added the dependency lately to the deb PPA file and exclude them here.

Reply:
with default sans serif LateNight looks sh**ty, labels spill out of buttons, layout affected etc.

where do we list dependencies? The fonts are not in README.md, also we don't pack the fonts in our ppa package -- which is where most users will get 2.3 from I suppose.

Reply:
> They are installed by default, so no one will notice

Ubuntu probably is, but at least here Open Sans was not, I had to get it manually.
will check Ubuntu Studio LiveUSB..

Reply:
Here: 
https://github.com/mixxxdj/mixxx/blob/68d291b20ec818984b9617aa882478414eea4b9d/CMakeLists.txt#L2654

Unfortunately every package manager needs to do this manually. 

Maybe we should centralize this.

Reply:
okay when installing 2.3 from the betas ppa I'm asked to install the fonts  :relieved:

Reply:
Thanks for the hint. I just added the missing `fonts-open-sans` dependency for Fedora.

Reply:
Do We have a conclusion here? I still think that we need at least some obvious documentation, if we decided not to install the fonts folder on Linux, else I am afraid this is a hard to see trap.

It should be clear for the package maintainers how to deal with fonts, that no font is missing and Mixxx looks ugly.  

An easy solution would be to add a readme file to the fonts folder, that says: "not installed on Linux" or we install it and use the file to silence package linting. 
We may also use a Glob expression to install the font folder only if there is something new to install.

What do you think?

Reply:
I think we should list all dependencies in the README or INSTALLING file (or we add a comment to CMakeLists.txt?), There we can also document what is installed and what isn't.

Reply:
Oh wait, I just have noticed that we have essentialy the same issue like with the udev file here. If we don't install the fonts here on Linux they are available when running from the source tree, but not after ```make install```.

Should we add them to the tools/debian_buildenv.sh? I am unsure.

I am against maintaining an extra file in addition that list install dependencies. Because this will likely be in outdated quickly and is a nother file to maintain.

How about using the same pattern as with the udev file, to install it by default and allow to disable it for packaging?

This is IMHO the best solution.

What do you think?

Reply:
> Should we add them to the tools/debian_buildenv.sh? I am unsure.

I added them, even though they are not really needed for compiling Mixxx. Let's not make it unnecessarily hard for packagers to get an idea what dependencies are needed. That would be solved with documenting them explicitly, but I'm also okay with just pointing them to the buildenv script instead.